versions/3.1.1.md

```diff
@@ -2346,6 +2346,13 @@ While composition offers model extensibility, it does not imply a hierarchy betw
 To support polymorphism, the OpenAPI Specification adds the `discriminator` keyword.
 When used, the `discriminator` indicates the name of the property that hints which schema definition is expected to validate the structure of the model.
 
+###### Generic (Template) Data Structures
+
+Implementations MAY support defining generic or template data structures using JSON Schema's dynamic referencing feature:
```

Comment:
Could you add a link to some "intro" material? I couldn't find a good starting point in https://json-schema.org/draft/2020-12/json-schema-core.

Reply:
@ralfhandl it's funny, I actually wrote a fairly enormous explanatory section and then cut it as I didn't want to bloat the section.  It's not trivial to explain.

The best explanation of this is @gregsdennis [post on the JSON Schema blog](https://json-schema.org/blog/posts/dynamicref-and-generics).  I'm not sure if we want to have a link to that in the OAS document?  Perhaps I could add a page to the learn site mentioning this and other JSON Schema techniques (including linking from the learn site to the JSON Schema blog?)

It's absolutely true that the JSON Schema spec is not clear on this, in part because we didn't realize `$dynamicRef` and `$dynamicAnchor` would support this when we came up with it.  Also, I have regrets about the writing in that whole section... many regrets...

Does the learn site idea sound good?  If so I will open a PR for that page and we can wait until we agree on it to merge this.  If not, I can try to add back the explanatory material here and we can figure out how much is worth including.  Or any combination of these approaches.

Reply:
@ralfhandl so it turns out I already wrote a Learn page on dynamic references that links to the JSON Schema blog posts on generics.  I've updated the PR to link to that Learn page.  And also note up front that there is an example in the Schema Object Examples section as otherwise it seems a little abrupt.  Does that work for you?

Reply:
I'm a bit torn on directly linking the Learn page without mentioning it as an "Informative reference" in the appendix.

On the other hand there is precedence for direct linking already in v3.1.0, for example the links to https://yaml.org/spec/1.2/spec.html.

Reply:
That means we only have to figure out where in the build process this gets added, and ideally also produce an appendix for "Informative references" without changing the spec sources.

Reply:
@ralfhandl please feel free to file an issue for that and add it to the "Automation & Infrastructure" GitHub project.  I'd prefer that we not hold up this PR based on the references appendx, as the concerns are orthogonal.

Reply:
Technically, only the markdown is normative, so our normative representation doesn't have that appendix at all.  Therefore it should not block anything.

Reply:
#3740